Source/WebKit/UIProcess/_WKTouchEventGenerator.mm

```diff
@@ -249,38 +248,39 @@ - (BOOL)_sendMarkerHIDEventInWindow:(UIWindow *)window completionBlock:(void (^)
     return YES;
 }
 
-- (void)_updateTouchPoints:(CGPoint*)points count:(NSUInteger)count window:(UIWindow *)window
+- (void)_updateTouchPoints:(std::span<CGPoint>)points window:(UIWindow *)window
 {
     HandEventType handEventType;
     
     // The hand event type is based on previous state.
     if (!_activePointCount)
         handEventType = HandEventTouched;
-    else if (!count)
+    else if (points.empty())
         handEventType = HandEventLifted;
-    else if (count == _activePointCount)
+    else if (points.size() == _activePointCount)
         handEventType = HandEventMoved;
     else
         handEventType = HandEventChordChanged;
     
     // Update previous count for next event.
-    _activePointCount = count;
+    _activePointCount = points.size();
 
     // Update point locations.
-    for (NSUInteger i = 0; i < count; ++i)
+    for (size_t i = 0; i < points.size(); ++i)
         _activePoints[i].point = points[i];
```

Comment:
I think @weinig showed me a slightly more elegant style for iterating a pair of arrays in a case like this with a modern for loop without having to use an index, maybe by zippering the two arrays together? Would love to use that if it’s elegant and memorable enough. Comes up 3 or 4 times in this pull request.

Reply:
I found:
```
std::array container1 = { 1, 2, 3, 4, 5  };
std::array container2 = { 6, 7, 8, 9, 10 };

for(auto& [a, b] : std::views::zip(container1, container2))
    // Do something.
```

but I think that requires C++23 (https://en.cppreference.com/w/cpp/ranges/zip_view). I'm not familiar with a nicer way to do this in C++20.

Reply:
If we can’t use this yet in WebKit due to what is available in the libraries and compilers, I understand. Once we can use it, I think that style will be much better than indexing, much harder to get wrong.

Reply:
We can also add a simple `zip` to WTF and use it while waiting to move our minimum C++ library version forward. My research indicates it’s straightforward, so doesn’t really have to be in the standard library.

Reply:
Not quite as nice as zip in this use case, but I recently added:

```
for (auto [ i, point ] : IndexedRange(points)) {
    _activePoints[i].point = point;
}
```

Reply:
Ok, following up in https://github.com/WebKit/WebKit/pull/36063